src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -0,0 +1,212 @@
+
+  private static final Pattern INDEXED_PLACEHOLDER_PATTERN = Pattern.compile("(?:\\[(\\d+)\\])");
+
+  // This holds the name of the known formatting functions and position of the string to be formatted
+  private static final Map<String, Integer> FORMATTING_FUNCTIONS = ContainerUtil.newHashMap(
+    Pair.pair("printf", 0),
+    Pair.pair("sprintf", 0),
+    Pair.pair("errorf", 0),
+    Pair.pair("fprintf", 1),
+    Pair.pair("fscanf", 1),
+    Pair.pair("scanf", 0),
+    Pair.pair("sscanf", 1),
+    Pair.pair("fatalf", 0),
+    Pair.pair("panicf", 0));
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitCallExpr(@NotNull GoCallExpr o) {
+        PsiReference psiReference = o.getExpression().getReference();
+
+        if (psiReference == null) {
+          return;
+        }
+
+        PsiElement resolved = psiReference.resolve();
+
+        if (!(resolved instanceof GoFunctionOrMethodDeclaration)) {
+          return;
+        }
+
+        if (!isFormattingFunction((GoFunctionOrMethodDeclaration)resolved)) return;
+        if (!isInCorrectPackage((GoFunctionOrMethodDeclaration)resolved)) return;
+
+        String name = getName((GoFunctionOrMethodDeclaration)resolved);
+        Integer placeholderPosition = FORMATTING_FUNCTIONS.get(name);
+        List<GoExpression> arguments = o.getArgumentList().getExpressionList();
+        if (arguments.size() <= placeholderPosition) {
+          return;
+        }
+
+        int parameterCount = arguments.size() - placeholderPosition - 1;
+        int placeholdersCount = getPlaceholders(arguments.get(placeholderPosition));
+        if (placeholdersCount == -1) {
+          return;
+        }
+
+        if (placeholdersCount != parameterCount) {
+          String message = String.format("Got %d placeholder(s) for %d arguments(s)", placeholdersCount, parameterCount);
+
+          PsiElement element = o.getExpression();
+          if (placeholdersCount < parameterCount) {
+            element = arguments.get(placeholderPosition);
+          }
+          holder.registerProblem(element, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+        }
+      }
+    };
+  }
+
+  private static boolean isInCorrectPackage(@NotNull GoFunctionOrMethodDeclaration element) {
+    GoFile containingFile = element.getContainingFile();
+    GoPackageClause filePackage = containingFile.getPackage();
+    if (filePackage == null ||
+        filePackage.getIdentifier() == null || containingFile.getImportPath() == null) {
+      return false;
+    }
+
+    String packageName = filePackage.getIdentifier().getText();
+    String packageImportPath = containingFile.getImportPath();
+
+    return (packageName.equals("fmt") && packageImportPath.equals("fmt")) ||
```

Comment:
it's easier to check "fmt".equals(packageImportPath), then you don't have to check whether packageImportPath is null